provider/application.go

```diff
@@ -0,0 +1,191 @@
+package provider
+
+import (
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/jgramoll/terraform-provider-spinnaker/client"
+)
+
+type awsProviderSettings struct {
+	UseAmiBlockDeviceMappings bool `mapstructure:"use_ami_block_device_mappings"`
+}
+
+type providerSettings struct {
+	AWS *[]*awsProviderSettings `mapstructure:"aws`
+}
+
+// Application deploy application in application
+type application struct {
+	ID                             string               `mapstructure:"id"`
+	Name                           string               `mapstructure:"name"`
+	Email                          string               `mapstructure:"email"`
+	RepoType                       string               `mapstructure:"repo_type"`
+	RepoProjectKey                 string               `mapstructure:"repo_project_key"`
+	RepoSlug                       string               `mapstructure:"repo_slug"`
+	CloudProviders                 []string             `mapstructure:"cloud_providers"`
+	ProviderSettings               *[]*providerSettings `mapstructure:"provider_settings"`
+	InstancePort                   int                  `mapstructure:"instance_port"`
+	PlatformHealthOnly             bool                 `mapstructure:"platform_health_only"`
+	PlatformHealthOnlyShowOverride bool                 `mapstructure:"platform_health_only_show_override"`
+	EnableRestartRunningExecutions bool                 `mapstructure:"enable_restart_running_executions"`
+}
+
+func (a *application) toClientApplication() *client.Application {
+	return &client.Application{
+		Name:                           a.Name,
+		Email:                          a.Email,
+		CloudProviders:                 strings.Join(a.CloudProviders, ","),
+		RepoType:                       a.RepoType,
+		RepoProjectKey:                 a.RepoProjectKey,
+		RepoSlug:                       a.RepoSlug,
+		ProviderSettings:               a.toClientProviderSettigns(a.ProviderSettings),
+		InstancePort:                   a.InstancePort,
+		PlatformHealthOnly:             a.PlatformHealthOnly,
+		PlatformHealthOnlyShowOverride: a.PlatformHealthOnlyShowOverride,
+		EnableRestartRunningExecutions: a.EnableRestartRunningExecutions,
+	}
+}
+
+func (a *application) toClientProviderSettigns(settings *[]*providerSettings) *client.ProviderSettings {
+	if settings != nil || len(*settings) > 0 {
+		for _, setting := range *settings {
+			if setting.AWS != nil && len(*setting.AWS) > 0 {
+				for _, aws := range *setting.AWS {
+					return &client.ProviderSettings{
+						AWS: &client.AwsProviderSettings{
```

Comment:
Can there be multiple AWS providers for an application?

Reply:
As I see on spinnaker documentation no. It's not  but I will double check